src/rust/cryptography-x509/src/common.rs

```diff
@@ -301,9 +301,120 @@ impl<'a> asn1::SimpleAsn1Writable for UnvalidatedVisibleString<'a> {
     }
 }
 
+/// Like `UnvalidatedIA5String`, but preserves the invariant that the
+/// underlying string is ASCII only.
+#[derive(Debug, PartialEq)]
+pub struct IA5String(String);
+
+impl IA5String {
+    pub(crate) fn new(value: String) -> Option<Self> {
+        if value.is_ascii() {
+            Some(Self(value))
+        } else {
+            None
+        }
+    }
+
+    pub(crate) fn as_str(&self) -> &str {
+        &self.0
+    }
+}
```

Comment:
This is really the same as `asn1::IA5String`, except that one is not owned. Is there a reason we need an owned type here? I don't see where in teh API we've discussed it'd be required.

Reply:
Actually, thinking about it more, it really should be case-preserving and the comparison itself should be case insensitive. Doing that now.

Reply:
Done.